net2cog/netcdf_convert.py

```diff
@@ -62,51 +62,62 @@ def _write_cogtiff(
         example :/home/dockeruser/converter/podaac/netcdf_converter/temp/
             netcdf_converter/
             RSS_smap_SSS_L3_8day_running_2020_037_FNL_v04.0_test
-    nc_xarray : xarray.Dataset
-        xarray dataset loaded from NetCDF file
-    variable_name: str
-        Name of the variable within the file to convert.
+    nc_xarray : xarray.DataTree
+        xarray DataTree loaded from NetCDF file. This represents the whole
+        file.
+    variable_path: str
+        Full of the variable within the file to convert.
     logger : logging.Logger
         Python Logger object for emitting log messages.
 
     Notes
     -----
     - Assumption that 0 is always on the prime meridian/equator.
-    - The output name for converted GeoTIFFs is `<variable name>.tif`, with any
+    - The output name for converted GeoTIFFs is `<variable path>.tif`, with any
       slashes replaced with underscores.
     """
 
-    logger.debug("NetCDF Var: %s", variable_name)
+    logger.debug("NetCDF Var: %s", variable_path)
 
-    if variable_name in EXCLUDE_VARS:
-        logger.debug(f"Variable {variable_name} is excluded. Will not produce COG")
+    if variable_path in EXCLUDE_VARS:
+        logger.debug(f"Variable {variable_path} is excluded. Will not produce COG")
         return None
 
-    output_basename = f'{variable_name}.tif'.replace('/', '_')
+    output_basename = f'{variable_path}.tif'.lstrip('/').replace('/', '_')
```

Comment:
The `lstrip` is because the `variable_path` has a leading slash, which meant the temporary files created had a leading `_`. Not a big deal as those are temporary files, but just looked a bit wonky.